app/src/main/kotlin/com/wire/android/ui/home/HomeScreen.kt

```diff
@@ -58,23 +65,51 @@ fun HomeScreen(startScreen: String?, viewModel: HomeViewModel) {
-                Box {
-                    val startDestination = HomeNavigationItem.all.firstOrNull { startScreen == it.route }?.route
-
-                    HomeNavigationGraph(
-                        homeState = homeState,
-                        navController = navController,
-                        startDestination = startDestination
-                    )
-
-                    HomeTopBar(
-                        title = currentNavigationItem.title,
-                        isSearchable = currentNavigationItem.isSearchable,
-                        scrollPosition = scrollPosition,
-                        onUserProfileClick = { viewModel.navigateToUserProfile() },
-                        onHamburgerMenuItemCLick = { openDrawer() }
-                    )
+            val homeNavigationGraph = @Composable {
+                val startDestination = HomeNavigationItem.all.firstOrNull { startScreen == it.route }?.route
+
+                HomeNavigationGraph(
+                    homeState = homeState,
+                    navController = navController,
+                    startDestination = startDestination
+                )
+            }
+
+            val homeTopBar = @Composable {
+                WireCenterAlignedTopAppBar(
+                    title = stringResource(id = currentNavigationItem.title),
+                    onNavigationPressed = { openDrawer() },
+                    navigationIconType = NavigationIconType.Menu,
+                    actions = {
+                        UserProfileAvatar(avatarUrl = "", status = UserStatus.AVAILABLE) {
+                            viewModel.navigateToUserProfile()
+                        }
+                    },
+                    elevation = if (currentNavigationItem.isSearchable) 0.dp else dimensions().topBarElevationHeight,
+                )
+            }
+
+            val homeContent = @Composable {
+                with(currentNavigationItem) {
+                    if (isSearchable) {
+                        AppTopBarWithSearchBarLayout(
+                            scrollPosition = scrollPosition,
+                            searchBarHint = stringResource(R.string.search_bar_hint, stringResource(id = title).lowercase()),
+                            //TODO: implement the search for home once we work on it, for now we do not care
+                            searchQuery = "",
+                            onSearchQueryChanged = {},
+                            onSearchClicked = { },
+                            onCloseSearchClicked = {},
+                            appTopBar = homeTopBar,
+                            content = {
+                                homeNavigationGraph()
+                            }
+                        )
+                    } else {
+                        Scaffold(topBar = homeTopBar) {
+                            homeNavigationGraph()
+                        }
+                    }
```

Comment:
Can we extract these variables definitions to separated functions to improve readability?

Reply:
sure ! :)